### ISSUE TITLE
Shift key request as custom extra-keys   v0.66

### ISSUE THREAD

**Issue by pattex16** [termux/termux-app#1038]
I know that since v0.66 you can edit the extra keys row as exposed in https://wiki.termux.com/wiki/Touch_Keyboard.
I immediately came back to the defaut v0.65 layout : ( extra-keys = [['ESC','/','-','HOME','UP','END','PGUP'],['TAB','CTRL','ALT','LEFT','DOWN','RIGHT','PGDN']]  ).
nonetheless i couldnt set a key as shift.
i've tried 'SHIFT' or 'SHFT' or 'LSHIFT' but none of theese work.

is shift implemented and if it is what string is it mapped to?

**Comment by Quasic** [comment]
Shift seems more complicated to add, and I don't know if keyboards are able to tell if an outside shift key is pressed by any standard means, so it will show lowercase letters when this shift is pressed for uppercase... Maybe it's not implemented for reasons like these?

**Comment by xeffyr** [comment]
@Quasic On my keyboard a key for "uppercase characters" is interpreted as shift in Termux. If this works for all keyboards, then we don't need separate shift key at all.

**Comment by pattex16** [comment]
> @Quasic On my keyboard a key for "uppercase characters" is interpreted as shift in Termux. If this works for all keyboards, then we don't need separate shift key at all.

Yeah that modifier works good for shortcuts like SHIFT + 'i' when using vim, however some cli software requires the actual key combinations that check for keyboard input.
For example in gboard when pressing the shift modifier, the top number row becomes a row of symbol keys.
That's why i think it would be interesting to have it separated

**Comment by Quasic** [comment]
There is a PR #417 to add SHIFT

**Comment by pattex16** [comment]
@Quasic Awesome!

**Comment by rcmosher** [comment]
I'd find 'SHIFT' very useful to do things like Shift+Tab, which you can't do with a normal mobile keyboard.

**Comment by chumaltd** [comment]
This feature is great!
Anyone can add next line to https://wiki.termux.com/wiki/Touch_Keyboard ?

```
*SHIFT ("special key")
```

Wiki edit requires Administrators privilege.

**Comment by Grimler91** [comment]
@chumaltd added, thanks!

**Comment by chumaltd** [comment]
@Grimler91 Nice work! Thanks a lot.
